rest/route/pod_agent_test.go

```diff
@@ -266,13 +290,44 @@ func TestPodAgentNextTask(t *testing.T) {
 			resp := rh.Run(ctx)
 			assert.Equal(t, http.StatusNotFound, resp.Status())
 		},
+		"RunShouldExitWithNonRunningPod": func(ctx context.Context, t *testing.T, rh *podAgentNextTask, env evergreen.Environment) {
+			p := getPod()
+			p.Status = pod.StatusStarting
+			require.NoError(t, p.Insert())
+			rh.podID = p.ID
+
+			resp := rh.Run(ctx)
+			assert.Equal(t, http.StatusOK, resp.Status())
+			nextTaskResp := resp.Data().(*apimodels.NextTaskResponse)
+			assert.Equal(t, nextTaskResp.ShouldExit, true)
+		},
+		"RunCorrectlyMarksContainerTaskDispatched": func(ctx context.Context, t *testing.T, rh *podAgentNextTask, env evergreen.Environment) {
+			p := getPod()
+			require.NoError(t, p.Insert())
+			d := getDispatcher()
+			require.NoError(t, d.Insert())
+			proj := getProject()
+			require.NoError(t, proj.Insert())
+			tsk := getTask()
+			require.NoError(t, tsk.Insert())
+			rh.podID = p.ID
+
+			resp := rh.Run(ctx)
+			assert.Equal(t, http.StatusOK, resp.Status())
+			nextTaskResp := resp.Data().(*apimodels.NextTaskResponse)
+			assert.Equal(t, nextTaskResp.ShouldExit, false)
+			assert.Equal(t, nextTaskResp.TaskId, "t1")
+			foundTask, err := task.FindOneId("t1")
+			require.NoError(t, err)
+			assert.Equal(t, foundTask.Status, evergreen.TaskDispatched)
+		},
```

Comment:
Should there be tests for next task behavior when the pod has already been dispatched a task to run, like:
* Pod has already been assigned a task that is still dispatchable and it simply return that task.
* Pod has already been assigned a task but it's no longer dispatchable, so the route tries to assign it a new task.

Reply:
Added a test for the first case but the second case has been removed in the route per [this comment](https://github.com/evergreen-ci/evergreen/pull/5750#discussion_r911277236)